config/application.php

```diff
@@ -63,15 +63,13 @@
 Config::define('DB_COLLATE', '');
 $table_prefix = env('DB_PREFIX') ?: 'wp_';
 
-if (env('DATABASE_URL')) {
-    $database_dsn = parse_url(env('DATABASE_URL'));
-    $db_name = substr($database_dsn['path'], 1);
-    $db_host = isset($database_dsn['port']) ? $database_dsn['host'] . ":" . $database_dsn['port'] : $database_dsn['host'];
+if (env('DB_URL')) {
+    $dsn = (object) parse_url(env('DB_URL'));
```

Comment:
I don't see the point of using implicit cast from an array to an object. Plus, I don't know if it's a good practice to do that 🤔

Reply:
Also, it add negative performance to do such casting.
We talking about microseconds, but, keep in mind that application.php is called upon *every* request.
So that microseconds penalty will apply to all requests. And I personally want to avoid that when/if possible.

Reply:
May I have your feedback on the performance problematic here @austinpray && @knowler ?

Reply:
I think looking at it from a performance prospective is incredibly silly. Every database query processor in Laravel casts to objects, not to mention eloquent. In this case, it is a single array. If we were pinching on performance to this extent, dotenv as a whole would be catastrophic, @austinpray's [wp-config](https://github.com/roots/wp-config) would be excessive overhead, heck, even using `parse_url()` is a huge performance impact. According to [this publication](https://books.google.ca/books?id=mCpnlNYzqOQC&pg=PT732&lpg=PT732&dq=is+parse_url+faster+than+preg_match%3F&source=bl&ots=qzZXezlo-6&sig=KX8areZ0ht01SrN3g_PmDi_IZEQ&hl=en&sa=X&ei=syeWUvDgE4zWoASw04KoAQ#v=onepage&q=parse_url&f=false) – doing it with `preg_match()` is 7.5x faster than `parse_url()`.

I think it comes down to readability and what feels more appropriate. I felt like this was a better fit in both cases.

Reply:
While, I do like working with objects since they are syntactically nicer to write — especially for string concatenations — I don’t think the pay off is much since we’re only using it for a few lines within an if clause that nobody will ever touch again. In this case, I’d vote performance over syntactical sugar since, like you said, this is called on every request. We should strive to keep the core as fast as possible, even if it’s a matter of microseconds.

Reply:
Really good points @Log1x — your comment didn't show up until after I posted mine. Principally, I still hold to what I said.

Reply:
Rofl can't believe I am even doing this

https://3v4l.org/Kkrc9/perf#output vs https://3v4l.org/FlTqR

`$dsn = (object) parse_url('postgres://AAAAAA:BBBBBBBBBBBBBBBBBBBB@CCCCCCCCCCCCCCCCCCCCCCCCCC.compute-1.amazonaws.com:5432/ddddddddd');` vs `$dsn = parse_url('postgres://AAAAAA:BBBBBBBBBBBBBBBBBBBB@CCCCCCCCCCCCCCCCCCCCCCCCCC.compute-1.amazonaws.com:5432/ddddddddd');`

| Trial | PHP Version | User Time (s) | 
| --- | --- | --- |
| (object) | 7.3 | 0.009 |
| (object) | 7.3 | 0.009 |
| (object) | 7.3 | 0.009 |
| no cast | 7.3 | 0.007 |
| no cast | 7.3 | 0.007 |
| no cast | 7.3 | 0.007 |
| (object) | 7.2 | 0.008 |
| (object) | 7.2 | 0.008 |
| (object) | 7.2 | 0.008 |
| no cast | 7.2 | 0.006 |
| no cast | 7.2 | 0.006 |
| no cast | 7.2 | 0.006 |
| (object) | 7.1 | 0.017 |
| (object) | 7.1 | 0.017 
| (object) | 7.1 | 0.017 |
| no cast | 7.1 | 0.003 |
| no cast | 7.1 | 0.003 |
| no cast | 7.1 | 0.003 |

7.1 takes a hit, but that is about to be EOLed this year. Nobody should be starting a new bedrock project on 7.1, why would they?

I say just roll with (object), it looks nicer.

Reply:
@swalkinshaw tie breaker please haha

Reply:
I do agree that performance impact are very very very tiny, but only for >PHP 7.1 setup. The question was revelant as the code manipulation is only needed there. 

Thanks for your feedback

Reply:
For posterity: 

If you were skeptical about my above table's results: you were correct to think so. The "User Time (s)" is off by a couple orders of magnitude.

Here's a better benchmark: https://gist.github.com/austinpray/ed8e2aab4b757f1e41d5298584eb1590 / https://docs.google.com/spreadsheets/d/1_MAFd_pxs333myM8WPsOnzBb0hh4ZtvJ5r2zj2HAfJw/edit?usp=sharing

![image](https://user-images.githubusercontent.com/2192970/52539977-f8b53b00-2d49-11e9-839e-e773d41cb19b.png)

So yeah there's a difference in php 7.1 but 7.2+ there is literally no difference whatsoever.